src/libraries/CurrencyDelta.sol

```diff
@@ -9,8 +9,8 @@ library CurrencyDelta {
     /// @notice calculates which storage slot a delta should be stored in for a given account and currency
     function _computeSlot(address target, Currency currency) internal pure returns (bytes32 hashSlot) {
         assembly ("memory-safe") {
-            mstore(0, target)
-            mstore(32, currency)
+            mstore(0, and(target, 0xffffffffffffffffffffffffffffffffffffffff))
```

Comment:
e.g. when someone passes in a `recipient` in `settle(recipient)` that `recipient` becomes the `target` here. They could pass in a dirty parameter

Reply:
this is only because of settle(recipient) though I believe? no where else in our code path could that have been the case no?

maybe if someone passed in a bad hook address, but then that hook wouldnt have even been useable

Reply:
i havent dug into this exact case - but generally certoras rule was that it was better to be careful and do it everywhere necessary instead of relying on things being safe. Especially as theres a chance we will change something about the code later and then miss cleaning stuff up.

Reply:
I think solidity already does this `AND` for all `address` parameters.

Reply:
> certoras rule was that it was better to be careful

It's either safe or it's not. If they don't know, then they have no business making recommendations. You should demand a refund.

Reply:
(and revert this PR)